src/algorithms/synthesis/syrec_synthesis.cpp

```diff
@@ -0,0 +1,1447 @@
+                    lines.emplace_back(offset + i);
+                }
+            }
+        } else {
+            for (unsigned i = 0u; i < var->var()->bitwidth(); ++i) {
+                lines.emplace_back(offset + i);
+            }
+        }
+        return true;
+    }
+
+    bool standard_syrec_synthesizer::unget_variables(const applications::variable_access::ptr& var, std::vector<unsigned>& lines [[maybe_unused]]) {
+        if (!var->indexes().empty()) {
+            // change offset
+
+            // check if it is not all numeric_expressions
+            unsigned n = var->var()->dimensions().size(); // dimensions
+            if ((unsigned)std::count_if(var->indexes().cbegin(), var->indexes().cend(), [&](const auto& p) { return dynamic_cast<applications::numeric_expression*>(p.get()); }) != n) {
+            }
+        }
+
+        return true;
+    }
+
+    /**
+   * Function to access array variables
+   *
+   * The array variable that corresponds to the given indexes is exchanged (via swap operations) with some given helper lines
+   *
+   * \param offset is the first line number associated to the array
+   * \param dimensions is the dimensions of the array
+   * \param indexes is the indexes of the array
+   * \param bitwidth is the bitwidth of the variables within the array
+   * \param lines is the destination, where
+   */
+    unsigned standard_syrec_synthesizer::get_constant_line(bool value) {
+        unsigned const_line = 0u;
+
+        if (!free_const_lines_map[value].empty()) {
+            const_line = free_const_lines_map[value].back();
+            free_const_lines_map[value].pop_back();
+        } else if (!free_const_lines_map[!value].empty()) {
+            const_line = free_const_lines_map[!value].back();
+            free_const_lines_map[!value].pop_back();
+            append_not(*(get(boost::vertex_name, cct_man.tree)[cct_man.current].circ), const_line);
+        } else {
+            const_line = add_line_to_circuit(_circ, (std::string("const_") + std::to_string(value)), "garbage", value, true);
+        }
+
+        return const_line;
+    }
+
+    bool standard_syrec_synthesizer::get_constant_lines(unsigned bitwidth, unsigned value, std::vector<unsigned>& lines) {
+        boost::dynamic_bitset<> number(bitwidth, value);
+
+        for (unsigned i = 0u; i < bitwidth; ++i) {
+            lines.emplace_back(get_constant_line(number.test(i)));
+        }
+
+        return true;
+    }
+
+    void standard_syrec_synthesizer::add_variable(circuit& circ, const std::vector<unsigned>& dimensions, const applications::variable::ptr& var,
+                                                  constant _constant, bool _garbage, const std::string& arraystr) {
+        if (dimensions.empty()) {
+            for (unsigned i = 0u; i < var->bitwidth(); ++i) {
+                std::string name = boost::str(boost::format(variable_name_format) % var->name() % i % arraystr);
```

Comment:
if the `variable_name_format` member is removed, this can be adapted to not use boost.